### PR TITLE
Functional tests - add test 'Preview Email Themes'

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add new functional test : **Preview Email Themes**
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `TEST_PATH="functional/BO/design/emailThemes/01_previewEmailThemes" URL_FO=Shop_Url npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17009)

### CODE REVIEW

File: tests/puppeteer/pages/BO/BObasePage.js
Comment (jolelievre) [inline tests/puppeteer/pages/BO/BObasePage.js:68]: There is a pending PR about this https://github.com/PrestaShop/PrestaShop/pull/16696
`AdminMailThemeParent ` is gonna be renamed into `AdminParentMailTheme`
Do you prefer to merge this one as is and fix it once the bugfix PR will be merged, or wait for the proper name to be available?

File: tests/puppeteer/pages/BO/design/emailThemes/index.js
Comment (jolelievre) [inline tests/puppeteer/pages/BO/design/emailThemes/index.js]: Seems a bit unstable to rely on children order no? couldn't you use some class or id which would have less chance to be modified?

Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/design/emailThemes/index.js]: in this case there is only one table, so `table.grid-table` should do it.
but it would be better to have id on the block or a name like in the bloc translate emails `form[name='generate_mails']`.
do you think i should delete `${this.blockDiv.replace('%ID', 5)}` ?

File: tests/puppeteer/pages/BO/BObasePage.js
Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/BObasePage.js:68]: We can pass this one and and fixing the id after that

File: tests/puppeteer/pages/BO/design/emailThemes/index.js
Comment (jolelievre) [reply tests/puppeteer/pages/BO/design/emailThemes/index.js]: Indeed it's too bad there is no ID to target this specific table, so if we add one it won't work any more
But for now yes I think you should only use `table.grid-table` to select it, because the chances are high that we add/remove/switch a block in this page The chances of a new table are low, but when it happens we will add the appropriate IDs

Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/design/emailThemes/index.js]: done

**Review by jolelievre — APPROVED** [review]

**Pull request merged**
